### PR TITLE
this -> this security feature (in basics/arrays)

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: basics/arrays.md [hunk 1]
```diff
@@ -3,8 +3,8 @@
 There are two types of Arrays in D: **static** and **dynamic**
 arrays. Access to arrays of any kind is always bounds checked -
```
Comment (ZombineDev) [inline basics/arrays.md]: Bounds checks can be elided if the compiler can prove that their not necessary. E.g.
https://github.com/dlang/dmd/pull/5470
https://github.com/dlang/dmd/pull/3620
https://github.com/dlang/dmd/pull/4351

Maybe also worth pointing out that by default, even in release mode, `@safe` functions are boundschecked, unless the `-boundschecks=off` option is specified, e.g. `@safe` functions are special.

File: basics/arrays.md [hunk 2]
```diff
@@ -3,8 +3,8 @@
...
 a failed range check yields a `RangeError` which aborts the application.
-The brave can disable this with the compiler flag `-boundschecks=off` to squeeze
-the last cycles out of their binary.
+The brave can disable this security feature with the
```
Comment (JackStouffer) [inline basics/arrays.md]: It's not a security feature as much as it's a safety feature. Security comes from `@safe`-ty in D, not the other way around.

**Comment by wilzbach** [comment]
> Maybe also worth pointing out that by default, even in release mode, @safe functions are boundschecked, unless the -boundschecks=off option is specified, e.g. @safe functions are special.

This is one of the first pages and I am already unsure whether it's good to point out that the DMD can elide bounds-checking, so I would leave this technical detail to the documentation.

File: basics/arrays.md [hunk 3]
```diff
@@ -1,10 +1,12 @@
 # Arrays
 
 There are two types of Arrays in D: **static** and **dynamic**.
-Access to arrays of any kind is always bounds checked -
-a failed range check yields a `RangeError` which aborts the application.
-The brave can disable this with the compiler flag `-boundschecks=off`, in order
-to squeeze the last cycles out of their binary.
+Access to arrays of any kind is bounds-checked (except the compiler can prove
```
Comment (wilzbach) [inline basics/arrays.md]: I think "bounds checked" deserves a hyphen here, see e.g.

http://www.apastyle.org/learn/faqs/when-use-hyphen.aspx

Comment (JackStouffer) [inline basics/arrays.md]: > except the compiler can prove that bound checks aren't necessary

Did you mean `except when the compiler can prove that bound checks aren't necessary`?

File: basics/arrays.md [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
+that bound checks aren't necessary).
+A failed range check yields a `RangeError` which aborts the application.
```
Comment (JackStouffer) [inline basics/arrays.md]: The term `range check` is introduced in this sentence with no introduction.

Would probably be clearer to say "A bounds check yields a `RangeError`"

File: basics/arrays.md [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
+The brave can disable this safety feature with the
+compiler flag `-boundschecks=off`
+in order to squeeze the last cycles out of their binary.
```
Comment (JackStouffer) [inline basics/arrays.md]: Would be clearer to say 

>The brave can disable this safety feature with the compiler flag `-boundschecks=off` in order to gain some speed at the cost of safety.

**Pull request merged**